Move PDP URLs from 'about-platform' to 'about'

zachtil opened the pull request:
## Describe this PR

The initial data-platform/about-platform URLs use twice the term platform which is easily avoided. However, it includes 3 changes:

* Changing the MDX file names
* Fixing links to /about-platform to now use /about
* Adding redirects (sigh)

## Changes

As described above.

janpio commented:
Technically the "platforms-releases" and "platform-limits-and-status" pages still duplicate "platform" 😆

zachtil commented:
> Technically the "platforms-releases" and "platform-limits-and-status" pages still duplicate "platform" 😆

Correct. So I've fixed those too. And added yet more redirects.

janpio commented on static/_redirects:
```diff
@@ -374,6 +374,14 @@
+/docs/data-platform/about-platform     /docs/data-platform/about   301!
+/docs/data-platform/about-platform/*   /docs/data-platform/about/:splat 301!
+
+/docs/data-platform/about-platform/platform-limits-and-status     /docs/data-platform/about/limits-and-status   301!
+/docs/data-platform/about-platform/platform-releases              /docs/data-platform/about/releases   301!
```
These two should never become active, as line 380 will catch these and redirect to `/docs/data-platform/about/:splat`.

So probably you want:
```suggestion
/docs/data-platform/about/platform-limits-and-status     /docs/data-platform/about/limits-and-status   301!
/docs/data-platform/about/platform-releases              /docs/data-platform/about/releases   301!
```

zachtil replied:
Thanks for catching this, Jan. That makes perfect sense.
